File: python/2221. Find Triangular Sum of an Array/solution.py
class Solution:
    def triangularSum(self, nums: List[int]) -> int:
        def calculateNextStep(nums):
            answer = [0] * (len(nums) - 1)

            for i in range(len(nums) - 1):
                answer[i] = (nums[i] + nums[i + 1]) % 10

            return answer


        while len(nums) != 1:
            nums = calculateNextStep(nums)

        return nums[0]

File: java/2221. Find Triangular Sum of an Array/Solution.java
class Solution {

    public int triangularSum(int[] nums) {
        if (nums.length == 1) return nums[0];

        int[] answer = new int[nums.length - 1];
        for (int i = 0; i < nums.length - 1; i++) {
            answer[i] = (nums[i] + nums[i + 1]) % 10;
        }

        return triangularSum(answer);

    }

    public int triangularSum(int[] nums) {
        while (nums.length != 1) {
            nums = calculateNextStep(nums);
        }
        return nums[0];
    }

    public int[] calculateNextStep(int[] nums) {
        int[] answer = new int[nums.length - 1];

        for (int i = 0; i < nums.length - 1; i++) {
            answer[i] = (nums[i] + nums[i + 1]) % 10;
        }

        return answer;
    }
}